components/hab/src/main.rs

```diff
@@ -869,7 +870,7 @@ fn sub_svc_set(m: &ArgMatches) -> Result<()> {
             .unwrap_or("UNKNOWN".to_string()),
     ))?;
     ui.status(Status::Creating, format!("service configuration"))?;
-    SrvClient::connect(&sup_addr, &secret_key)
+    SrvClient::connect(&sup_addr.into(), &secret_key)
```

Comment:
This makes me want to update `SrvClient::connect` to take an `AsRef<SocketAddr>` instead of an `&SocketAddr`, and then implement that trait for `ListenCtlAddr`. 

That would better than `Into<SocketAddr>` given [this API guideline](https://rust-lang-nursery.github.io/api-guidelines/interoperability.html#c-conv-traits)